Change pilight systemcode validation to integer

janLo opened the pull request:
**Description:**
This changes the config validation for systemcode in pilight-switch to be an integer instead of a string. Strings cause an error in the pilight daemon. The systemcode is used as integer everywhere in the pilight code:
https://github.com/pilight/pilight/search?utf8=%E2%9C%93&q=systemcode

**Related issue (if applicable):** fixes #4282 


**Checklist:**

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io)

If the code communicates with devices, web services, or third-party tools:
  - [ ] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [ ] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [ ] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [ ] New dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [ ] New files were added to `.coveragerc`.

If the code does not interact with devices:
  - [ ] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [ ] Tests have been added to verify that the new code works.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L16
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L51

According to the pilight code the systemcode should be an integer and
not a string (it is an int in the pilight code). Passing this as a
string caused errors from pilight:
"ERROR: elro_800_switch: insufficient number of arguments"

This fixes #4282

janLo committed:
Change pilight systemcode validation to integer

According to the pilight code the systemcode should be an integer and
not a string (it is an int in the pilight code). Passing this as a
string caused errors from pilight:
"ERROR: elro_800_switch: insufficient number of arguments"

This fixes #4282

DavidLP commented:
Generally I never liked that we define the `COMMAND_SCHEMA` [here](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/switch/pilight.py#L27), because it completely depends on the device, thus is just a general yaml string with all key/values possible. 
I would prefer to delete this, otherwise I already see bugs at the horizon where e.g. `systemcode` is missing but `unitkey` is defined and needs to be integer. I hope that it is still possible with the voluptuous changes to profit from python ducktyping.

janLo commented:
I think, this validation should be done properly in the pilight daemon and errors should be propagated back correctly to the clients.

For now, this PR fixes an regression introduced with version 0.32.

DavidLP commented:
E. G. CONF_UNIT also has to be also a positive integer. Thus as I said, no type check is better here, since we do not want to define all possible fields for all possible devices. But for this hot fix just also changing CONF_UNIT should be fine. At a later point I will partially revert this type checks in another pull request.

DavidLP commented:
[Here](https://community.home-assistant.io/t/pilight-component-broken-since-last-update/5879/6?u=davidlp) some discussion from users

janLo committed:
Change pilight unit-id to positive integer

According to the pilight code the unit of an entity is also evrywhere
handled as an integer. So converting and passing this as string causes
pilight not to work.

This fixes #4282

Signed-off-by: Jan Losinski <losinski@wh2.tu-dresden.de>

janLo commented:
Updated the unitcode to a positive int as the comments in the discussion above mentioned that this is also an mistake.

I'll try to get a real protocol-specific parameter validation out of the pilight code as soon as the time allows this. This can than either be used in the HA pilight component or in the pilight python module.
